backend/app/controllers/uoas.js

```diff
@@ -91,34 +88,95 @@ module.exports = {
         }
 
         co(function* () {
-            var added = yield thunkQuery(
-                'SELECT name, id FROM "UnitOfAnalysis" ' +
+            // Check if Subjects already exist in DB
+            const existingRecords = yield thunkQuery(
+                'SELECT * FROM "UnitOfAnalysis" ' +
                 'WHERE LOWER("UnitOfAnalysis"."name") IN (' + sqlString.toLowerCase() + ') ' +
```

Comment:
So this is my old derp, but the `sqlString` has a problem where it is oly declared if the body has subjects (the if clause on line 79). If you would do me a favor and move it above that if clause, say `let sqlString;`

Reply:
Maybe, I am not following fully, but why would we want to declare `sqlString` or `uoa` if there are no subjects? If there are no subjects, we just throw an error saying there are not subjects - line 87.

Reply:
Generally because it's declaring outside of a conditional scope, ensuring that the variable does exist.

Reply:
Done!